From 5d0031339ab287ea14966e00239c3c756a9c3ee6 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Thu, 6 Feb 2025 15:33:17 -0700
Subject: [PATCH] fix: Allow for ninja and collier to be shared libraries

* It is valid for external instances of ninja and collier to be shared libraries,
  and so accept them if they are found when a static libray version is not.
---
 madgraph/interface/madgraph_interface.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/madgraph/interface/madgraph_interface.py b/madgraph/interface/madgraph_interface.py
index ebc9572..1e545d6 100755
--- a/madgraph/interface/madgraph_interface.py
+++ b/madgraph/interface/madgraph_interface.py
@@ -7996,6 +7996,9 @@ in the MG5aMC option 'samurai' (instead of leaving it to its default 'auto')."""
                 self.options[args[0]] = None
             else:
                 program = misc.which_lib(os.path.join(args[1],'lib%s.a'%args[0]))
+                # Check for shared library
+                if program is None:
+                    program = misc.which_lib(os.path.join(args[1],'lib%s.so'%args[0]))
                 if program!=None:
                     res = 0
                     logger.info('set %s to %s' % (args[0],args[1]))
-- 
2.47.1

